### PR TITLE
Simplify player status string building

### PR DESCRIPTION
**Pull request by RoiEXLab**
Follow-up to #5289

## Functional Changes

[ ] New map or map update
[ ] New Feature
[ ] Feature update or enhancement
[ ] Feature Removal
[x] Code Cleanup or refactor
[ ] Configuration Change
[ ] Bug fix:  
[ ] Other:   

## Testing

[x] No manual testing done
[ ] Manually tested

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java [hunk 1]
```diff
@@ -238,24 +239,11 @@ private String getDisplayString(final ChatParticipant chatParticipant) {
...
-    final StringBuilder sb = new StringBuilder();
-    if (status.length() > 0) {
-      if (status.length() > 25) {
-        status = status.substring(0, 25);
-      }
-      for (int i = 0; i < status.length(); i++) {
-        final char c = status.charAt(i);
-        if (c >= '\u0300' && c <= '\u036F') { // skip combining characters
-          continue;
-        }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: Note that this PR drops this replacement.
However if there are any objections to this, I can just revert the latest commit to get a version that keeps this replacement.

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: We should understand it before dropping it; hopefully there was a good reason it was added to begin with. It's really annoying to see code like this, at least there was some commentary noted to help, but the exact logic is elusive.

File: game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java [hunk 2]
```diff
@@ -238,24 +239,11 @@ private String getDisplayString(final ChatParticipant chatParticipant) {
       return "";
     }
 
-    String extra = chatParticipant.isModerator() ? " " + TAG_MODERATOR : "";
-    String status = chat.getStatus(chatParticipant.getPlayerName());
+    final String extra = chatParticipant.isModerator() ? " " + TAG_MODERATOR : "";
+    final String status = Ascii.truncate(chat.getStatus(chatParticipant.getPlayerName()), 25, "");
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: Have you had any thoughts whether chat truncation be done at the client, just the server, or both server and client?

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: I believe @ssoloff dealt with sanitation before, maybe he knows something about it

File: game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java (hunk 1 above)
Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: Well I checked the unicode table of this particular range and it's like the comment states:
All of those characters are "combination chars" that give a ^ to a character. According to wikipedia those are called [diacritic signs](https://en.wikipedia.org/wiki/Diacritic) as a whole.

So removing those should really not have a noticeable effect.

File: game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java (hunk 2 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/chat/ChatPlayerPanel.java]: Sorry, got a bit rhetorical. The truncation probably should be done in chat controller, unfortunately chat messages are sent via broad cast, so they are blindly forwarded.

**Pull request merged**